woocommerce/woocommerce
# Add SortableList component

### All Submissions:






### Changes proposed in this Pull Request:

<img width="420" alt="Screen Shot 2022-08-08 at 4 17 59 PM" src="https://user-images.githubusercontent.com/10561050/183530533-4f17e255-7d2b-4533-bfdd-e2a5d25c38e6.png">


This PR adds a sortable list component to allow sorting of items and callbacks to retrieve the order on completion.  Note that this component should remain largely open-ended, but follow-ups or separate PRs will be created for the following:

* Design - the list item will drive more design than this component
* Drag tests - For some reason the dragging events are not firing within GB's `Draggable` component, making this a little harder to test.  We can follow up or implement here if anyone has input
* Accessibility - Keyboard events and better labeling will occur in a follow-up to keep the scope of this PR down

### How to test the changes in this Pull Request:

1. `npm run storybook`
2. Check the "SortabeList" component
3. Check that items can be dragged and dropped
4. Check that a callback occurs on reordering items (first storybook example)

### Other information:




### FOR PR REVIEWER ONLY:
